x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -935,9 +942,21 @@ export class TaskRunner<
         }
 
         return { interval: retryInterval };
-      }),
+      }
+    );
+
+    const getControlledError = (scheduleResult: Result<IntervalSchedule, ErrorWithReason>) => {
+      if (isErr(scheduleResult)) {
+        const { error, source } = scheduleResult.error;
+        return createTaskRunError(error, source);
+      }
+    };
+
+    return {
+      state: transformedState,
+      schedule: scheduleWithInterval,
       monitoring: this.ruleMonitoring.getMonitoring(),
-      hasError: isErr(schedule),
+      controlledError: getControlledError(schedule),
```

Comment:
As rule task runner always return success, we had added `hasError` flag pass error status to Task Manager.
With this POC, we replace it with `controlledError` that will be used by the Task manager to differentiate the errors.

Reply:
controlledError -> taskRunError